x-pack/test/api_integration/apis/asset_manager/tests/assets_hosts.ts

```diff
@@ -0,0 +1,172 @@
+  const es = getService('es');
+  const synthtraceEsClient = getService('synthtraceEsClient');
+
+  describe('asset management', () => {
+    describe('GET /assets/hosts', () => {
+      before(async () => {
+        synthtraceEsClient.pipeline(getInfraPipeline());
+      });
+
+      it('should return host assets from signals', async () => {
+        const end = new Date(); // now
+        const start = new Date();
+        start.setHours(end.getHours() - 1); // now-1h
+
+        const events = timerange(start.getTime(), end.getTime())
+          .interval('10m')
+          .rate(1)
+          .generator<InfraDocuments>((timestamp) => {
+            const hostA = infra.host('host-a');
+            const podA = hostA.pod('pod-a');
+
+            const hostB = infra.host('host-b');
+            const podB = hostA.pod('pod-b');
+
+            return [
+              hostA.metrics(),
+              hostA.asset(),
+              podA.metrics(),
+              podA.asset(),
+              hostB.metrics(),
+              hostB.asset(),
+              podB.metrics(),
+              podB.asset(),
+            ].map(withTimestamp(timestamp));
+          });
+
+        await synthtraceEsClient.index(events); // Only refreshes APM data streams at the end
+        await refreshInfraDataStreams(es);
+
+        const response = await supertest
+          .get(HOST_ASSETS_ENDPOINT)
+          .query({
+            from: start,
+            to: end,
+          })
+          .expect(200);
+        expect(response.body).to.have.property('hosts');
+        expect(response.body.hosts.length).to.equal(2);
+        expect(response.body.hosts.map(dropTimestamp)).to.eql([
+          {
+            'asset.kind': 'host',
+            'asset.id': ['host-a'],
+            'asset.name': ['host-a'],
+            'asset.ean': 'host:host-a',
+          },
+          {
+            'asset.kind': 'host',
+            'asset.id': ['host-b'],
+            'asset.name': ['host-b'],
+            'asset.ean': 'host:host-b',
+          },
+        ]);
+      });
+
+      afterEach(async () => {
+        await deleteInfraDataStreams(es);
+      });
+    });
+  });
+}
+
+// Can I type better than any here? Like Fields?
+function withTimestamp<T extends Serializable<any>>(timestamp: number) {
+  return (document: T) => document.timestamp(timestamp);
+}
+
+// Is this a good solution? Or would it be better to capture the synthtrace interval timestamps to use/predict?
+function dropTimestamp(asset: Asset) {
+  const copy: Partial<Asset> = Object.assign({}, asset);
+  delete copy['@timestamp'];
+  return copy as AssetWithoutTimestamp;
+}
+
+// I'll probably want to write/extend/refactor the synthtrace client to be not so APM bound
+function getInfraPipeline() {
+  return (base: Readable) => {
+    return pipeline(
+      base,
+      getSerializeTransform(),
+      getRoutingTransform(),
+      getDedotTransform(),
+      (err) => {
+        if (err) {
+          // What to do here?
```

Comment:
If a failure here breaks the later testing assumptions, because for example the expected data is only partially indexed, I'd vote to throw here

Reply:
Agreed!